relay-event-schema/src/protocol/span_v2.rs

```diff
@@ -0,0 +1,388 @@
+    fn is_empty(&self) -> bool {
+        false
+    }
+}
+
+#[derive(Debug)]
+pub struct ParseSpanKindError;
+
+impl std::str::FromStr for SpanV2Kind {
+    type Err = ParseSpanKindError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "unspecified" => Self::Unspecified,
+            "internal" => Self::Internal,
+            "server" => Self::Server,
+            "client" => Self::Client,
+            "producer" => Self::Producer,
+            "consumer" => Self::Consumer,
+            _ => return Err(ParseSpanKindError),
+        })
+    }
+}
+
+impl Default for SpanV2Kind {
+    fn default() -> Self {
+        Self::Internal
+    }
+}
+
+impl From<OtelSpanKind> for SpanV2Kind {
+    fn from(otel_kind: OtelSpanKind) -> Self {
+        match otel_kind {
+            OtelSpanKind::Unspecified => Self::Unspecified,
+            OtelSpanKind::Internal => Self::Internal,
+            OtelSpanKind::Server => Self::Server,
+            OtelSpanKind::Client => Self::Client,
+            OtelSpanKind::Producer => Self::Producer,
+            OtelSpanKind::Consumer => Self::Consumer,
+        }
+    }
+}
+
+impl fmt::Display for SpanV2Kind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.as_str())
+    }
+}
+
+impl FromValue for SpanV2Kind {
+    fn from_value(value: Annotated<Value>) -> Annotated<Self>
+    where
+        Self: Sized,
+    {
+        match value {
+            Annotated(Some(Value::String(s)), meta) => Annotated(Self::from_str(&s).ok(), meta),
+            Annotated(_, meta) => Annotated(None, meta),
+        }
+    }
+}
+
+impl IntoValue for SpanV2Kind {
+    fn into_value(self) -> Value
+    where
+        Self: Sized,
+    {
+        Value::String(self.to_string())
+    }
+
+    fn serialize_payload<S>(
+        &self,
+        s: S,
+        _behavior: relay_protocol::SkipSerialization,
+    ) -> Result<S::Ok, S::Error>
+    where
+        Self: Sized,
+        S: serde::Serializer,
+    {
+        s.serialize_str(self.as_str())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use chrono::{TimeZone, Utc};
+    use similar_asserts::assert_eq;
+
+    use super::*;
+
+    macro_rules! attrs {
+        ($($name:expr => $val:expr , $ty:ident),* $(,)?) => {
+            std::collections::BTreeMap::from([$((
+                $name.to_owned(),
+                relay_protocol::Annotated::new(
+                    $crate::protocol::Attribute::new(
+                        $crate::protocol::AttributeType::$ty,
+                        $val.into()
+                    )
+                )
+            ),)*])
+        };
+    }
+
+    #[test]
+    fn test_span_serialization() {
+        let json = r#"{
+  "trace_id": "6cf173d587eb48568a9b2e12dcfbea52",
+  "span_id": "438f40bd3b4a41ee",
+  "name": "GET http://app.test/",
+  "status": "ok",
+  "is_remote": true,
+  "kind": "server",
+  "start_timestamp": 1742921669.158209,
+  "end_timestamp": 1742921669.180536,
```

Comment:
This test doesn't quite work: deserializing these timestamps doesn't result in exactly the values further down, for reasons I don't understand.